elastic/beats
# Get rid of errors generated by xpack code for `elasticsearch/shard` metricset

I was testing the `elasticsearch/shard` metricset with `xpack.enabled: true` on `master` and ran into a few errors:

```
3 errors: wrong format in `relocating_node`: expected string, found <nil>; wrong format in `node`: expected string, found <nil>; key `number` not found"
```

Investigating a bit I found that:
1. `node` and `relocating_node` can sometimes be returned as `null` in the `GET _cluster/state/version,master_node,nodes,routing_table` Elasticsearch API response used by this metricset, and
2. the `number` field has been renamed in the Elasticserach API response to `shard` now. I can't find when it was `number` but it is definitely `shard` now.

This PR updates the code to get rid of these errors.

It also un-hardcodes the name of the monitoring index set in `data_xpack.go`.

### Testing this PR
1. Enable the `elasticsearch` metricset:
   ```
   metricbeat modules enable elasticsearch
   ```

2. Edit `modules.d/elasticsearch.yml` and set `metricsets: [ "shard" ]` and `xpack.enabled: true`.

3. Run metricbeat:
   ```
   metricbeat -e
   ```

4. Check in Elasticsearch that you don't have any documents in `metricbeat-*` for the `elasticsearch` module (since we are using `xpack.enabled: true`).
   ```
   GET metricbeat-*/_search?q=metricset.module:elasticsearch
   ```

5. Check in Elasticsearch that you have documents in `.monitoring-es-6-mb-*` with `type:shard`
   ```
   GET .monitoring-es-6-mb-2*/_search?q=type:shards
   ```
